cypress/integration/pages/mostReadPage/index.js

```diff
@@ -1,43 +1,21 @@
 import config from '../../../support/config/services';
-import getPaths from '../../../support/helpers/getPaths';
-import serviceHasPageType from '../../../support/helpers/serviceHasPageType';
 import testsForCanonicalOnly from './testsForCanonicalOnly';
 import crossPlatformTests from './tests';
 import visitPage from '../../../support/helpers/visitPage';
+import getAppEnv from '../../../support/helpers/getAppEnv';
 
 const pageType = 'mostReadPage';
-Object.keys(config)
-  .filter(service => serviceHasPageType(service, pageType))
-  .forEach(serviceId => {
-    // eslint-disable-next-line prefer-const
-    let { variant, name: service } = config[serviceId];
-    if (variant !== 'default') {
-      const capitalisedVariant =
-        variant.charAt(0).toUpperCase() + variant.slice(1);
-      service += capitalisedVariant;
-    }
-    const paths = getPaths(serviceId, pageType);
-    paths.forEach(currentPath => {
-      describe(`${pageType} - ${currentPath}`, () => {
-        before(() => {
-          Cypress.env('currentPath', currentPath);
-          visitPage(currentPath, pageType);
-        });
-        crossPlatformTests({
-          service,
-          pageType,
-          variant,
-        });
-        testsForCanonicalOnly({
-          service,
-          pageType,
-          variant,
-        });
-      });
-    });
-    paths
-      .map(path => `${path}.amp`)
-      .forEach(currentPath => {
+
+Object.values(config).forEach(
+  ({ name: service, variant, pageTypes, isWorldService }) => {
+    if (isWorldService) {
+      const { mostReadPage } = pageTypes;
+
+      const urls =
+        mostReadPage.environments &&
+        mostReadPage.environments[getAppEnv()]?.paths;
```

Comment:
**Note:** this tests all most read pages, regardless of whether they are enabled / disabled. This is to ensure that we always have correct most read data, which contains a title & a link

Reply:
If a most read page is disabled, would it have the list on it? Are there any empty most read pages due to disabling? 

I have also remembered that Most Read pages are supposed to be empty if there is not enough data to fill the list to 5 or 10 (depending on settings). But the part of the test that checks there are enough records is not longer there. Has the requirement changed and now there will always be the component on the page even if there are not enough records, or are the tests going to fail when they test a page that has a most read with not enough data to show the list?

Reply:
All most read pages should be able to render content, and the mostRead toggle is enabled for all services (so we could technically remove it now - I think it was used to roll out the component to each service when it was first built). 

From what I've seen, there does appear to be 20 records returned for each service, and the services either display the top 5 or 10 items from the list. 

There _could_ be a scenario whereby the mostRead API timed out (after 2 seconds), and there would be no data at all to display. In this case, the page would fail - but that's good because we would probably want to be made aware of that, and investigate